staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2728,6 +2730,11 @@ type MigrationConfiguration struct {
 	// If set to true, migrations will still start in pre-copy, but switch to post-copy when
 	// CompletionTimeoutPerGiB triggers. Defaults to false
 	AllowPostCopy *bool `json:"allowPostCopy,omitempty"`
+	// AllowWorkloadDisruption indicates that the migration shouldn't be
+	// canceled after acceptableCompletionTime is exceeded. Instead, if
+	// permitted, migration will be switched to post-copy or the VMI will be
+	// paused to allow the migration to complete
+	AllowWorkloadDisruption *bool `json:"allowWorkloadDisruption,omitempty"`
```

Comment:
The comment for (at least) `CompletionTimeoutPerGiB` needs to be updated to reflect the new logic (pause if `AllowWorkloadDisruption && !AllowPostCopy`). Maybe we can think of simple wording like "If the timeout is reached, the migration will be either paused, switched to post-copy or cancelled depending on other settings".